test/e2e/custom_route_test.go

```diff
@@ -0,0 +1,185 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+
+	e2e "github.com/openshift/cluster-authentication-operator/test/library"
+)
+
+func TestCustomRouterCerts(t *testing.T) {
+	kubeConfig := e2e.NewClientConfigForTest(t)
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	routeClient, err := routeclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	// generate crypto materials
+	rootCA := e2e.NewCertificateAuthorityCertificate(t, nil)
+	intermediateCA := e2e.NewCertificateAuthorityCertificate(t, rootCA)
+	clientCA := e2e.NewClientCertificate(t, intermediateCA, "foo.bar.com")
+
+	// create tls secret
+	var certificates bytes.Buffer
+	certificates.Write(pem.EncodeToMemory(&pem.Block{Type: cert.CertificateBlockType, Bytes: clientCA.Certificate.Raw}))
+	privateKey, err := keyutil.MarshalPrivateKeyToPEM(clientCA.PrivateKey)
+	require.NoError(t, err)
+	secret, err := kubeClient.CoreV1().Secrets("openshift-config").Create(
+		context.TODO(),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: strings.ReplaceAll(strings.ToLower(t.Name()), "/", " ") + "-"},
+			Type:       corev1.SecretTypeTLS,
+			Data: map[string][]byte{
+				"tls.crt": certificates.Bytes(),
+				"tls.key": privateKey,
+			},
+		}, metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer func() {
+		removeComponentRoute(t, configClient, "openshift-authentication", "oauth-openshift")
+		_ = kubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.TODO(), secret.Name, metav1.DeleteOptions{})
+	}()
+
+	// Set a custom hostname without a secret
+	err = getAndUpdateComponentRoute(t, configClient, &configv1.ComponentRouteSpec{
```

Comment:
The operator can enter a degraded state for a few minutes when moving towards custom routes because the expected metadata returned by the well-known endpoint isn't updated quick enough.

Reply:
this should now be fixed as https://github.com/openshift/cluster-authentication-operator/pull/442 merged